pkg/cache/cache.go

```diff
@@ -109,3 +116,29 @@ func (c *Cache) Shutdown() {
 	c.Redis.Close()
 	log.Info().Msg("Successfully closed Redis connection")
 }
+
+// GetCache attempts to retrieve and unmarshal data from cache
+func (c *Cache) GetCache(data CacheableData, value interface{}) error {
```

Comment:
in this function, having both `data` and `value` are ambiguous, it would be cleaner if first param is like`GetCache(key string, ...)` since expiration time is not used here